File: message_service/Database/Models/Message.js
const mongoose = require("mongoose");

/**
 * The schema for the message
 * 
 * @type {ObjectId} chatID
 * @type {ObjectId} sender_id
 * @type {ObjectId} recipient_id
 * @type {Date} timestamp
 * @type {String} content
 */
const MessageSchema = new mongoose.Schema({
    
    chatID: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Match',
        required: true
    },

    timestamp: {
        type: Date,
        required: true,
        default: Date.now
    },

    sender_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
    },

    recipient_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
    },

    content: {
        type: String,
        required: true,
    }
});

module.exports = {
    Message: mongoose.model("Message", MessageSchema)
};

File: my_app/lib/utils/ChatService/Message.dart
class Message {
  late String content;
  late String sender_id;
  late String recipient_id;
  late String chat_id;
  late DateTime time_sent;

  Message(
      {required this.content,
      required this.sender_id,
      required this.recipient_id,
      required this.chat_id,
      required this.time_sent});

  Message.fromJSON(Map<String, dynamic> json) {
    content = json["content"];
    sender_id = json["sender_id"];
    recipient_id = json["recipient_id"];
    chat_id = json["chatID"];
    time_sent = DateTime.parse(json["timestamp"]);
  }

  static JsonToList(List<dynamic> data) {
    List<Message> messages = [];
    for (int i = 0; i < data.length; i++) {
      messages.add(Message.fromJSON(data[i]));
    }
    return messages;
  }
}
